flax/errors.py

```diff
@@ -69,67 +69,82 @@ def __init__(self, message):
     error_msg = f'{message} ({error_page}#{module_name}.{class_name})'
     super().__init__(error_msg)
 
+
 #################################################
 # scope.py errors                               #
 #################################################
 
-class InitScopeInvalidRngsError(FlaxError):
+
+class MissingRngError(FlaxError):
+  """
+  This error is raised if a PRNG is required but it is not provided as an input.
+  Please see :class:`InvalidRngError` for more details.
   """
-  When initializing a Module with
-  :meth:`Module.init() <flax.linen.Module.init>`, the first argument can be of
-  two forms:
+  def __init__(self, scope_name, rng_name):
+    super().__init__(f'{scope_name} needs PRNG for "{rng_name}"')
 
-  1. A single PRNGKey. This is in case only one PRNGKey is needed to initialize
-     the ``params`` collection. Note that this::
 
-       SomeModule(...).init(jax.random.PRNGKey(0), ...)
+class InvalidRngError(FlaxError):
+  """
+  All rngs used in a Module should be passed to 
+  :meth:`Module.init() <flax.linen.Module.init>` and 
+  :meth:`Module.apply() <flax.linen.Module.apply>` appropriately. We explain
+  both separately using the following example::
 
-     Is shorthand for::
+    class Bar(nn.Module):
+      @nn.compact
+      def __call__(self, x):
+        some_param = self.param('some_param', nn.initializers.zeros, (1, ))
+        dropout_rng = self.make_rng('dropout')
+        x = nn.Dense(features=4)(x)
+        ...
 
-       SomeModule(...).init({'params': jax.random.PRNGKey(0)}, ...)
+    class Foo(nn.Module):
```

Comment:
You could just get rid of `class Foo`, and use `Bar` directly?

Reply:
I've done that to clarify that even if a Module has submodules, there is no nesting in the provided RNGS (the provided RNGs is always a flat dict). But if you think this is a bit artificial I can also simplify it.

Reply:
Sure, good point.